src/languages/mathematica.js

```diff
@@ -1,59 +1,130 @@
-Description: Wolfram Mathematica (usually termed Mathematica) is a modern technical computing system spanning most areas of technical computing.
-Authors: Daniel Kvasnicka <dkvasnicka@vendavo.com>, Jan Poeschko <jan@poeschko.com>
+Language: Wolfram Language
+Description: The Wolfram Language is the programming language used in Wolfram Mathematica, a modern technical computing system spanning most areas of technical computing.
+Authors: Patrick Scheibe <patrick@halirutan.de>, Robert Jacobson <robertjacobson@acm.org>
 Website: https://www.wolfram.com/mathematica/
 Category: scientific
 */
 
+/*
+  The ./lib/mathematica.js file was created with Wolfram Mathematica 12.1.1 under Linux
+  with the following code:
+  $outputFile="/path/to/highlight.js/src/languages/lib/mathematica.js";
+  getStrippedContextNames[context_String]:=Block[
+    {
+      $ContextPath={context}
+    },
+    Union[Names@RegularExpression@StringJoin[context,"\\$?[A-Z]\\w*"]]
+  ];
+  $symbols=getStrippedContextNames["System`"];
+  Export[$outputFile,
+    With[
+      {
+        quoted="\""<>#<>"\""&/@$symbols
+      },
+      StringTemplate["export const SYSTEM_SYMBOLS = [\n`syms`\n];\n"][
+        <|"syms"->StringRiffle[Map["  " <> #&,quoted],",\n"]|>
+      ]
+    ],"String"]
+*/
+import * as Mathematica from './lib/mathematica.js';
+import * as regex from '../lib/regex.js';
+
+// @ts-ignore
 export default function(hljs) {
+  /*
+  This rather scary looking matching of Mathematica numbers is carefully explained by Robert Jacobson here:
+  https://wltools.github.io/LanguageSpec/Specification/Syntax/Number-representations/
+   */
+  const BASE_RE = /([2-9]|[1-2]\d|[3][0-5])\^\^/;
+  const BASE_DIGITS_RE = /(\w*\.\w+|\w+\.\w*|\w+)/;
+  const NUMBER_RE = /(\d*\.\d+|\d+\.\d*|\d+)/;
+  const BASE_NUMBER_RE = regex.either(regex.concat(BASE_RE, BASE_DIGITS_RE), NUMBER_RE);
+
+  const ACCURACY_RE = /``[+-]?(\d*\.\d+|\d+\.\d*|\d+)/;
+  const PRECISION_RE = /`([+-]?(\d*\.\d+|\d+\.\d*|\d+))?/;
+  const APPROXIMATE_NUMBER_RE = regex.either(ACCURACY_RE, PRECISION_RE);
+
+  const SCIENTIFIC_NOTATION_RE = /\*\^[+-]?\d+/;
+
+  const MATHEMATICA_NUMBER_RE = regex.concat(
+    BASE_NUMBER_RE,
+    regex.optional(APPROXIMATE_NUMBER_RE),
+    regex.optional(SCIENTIFIC_NOTATION_RE)
+  );
+
+  const NUMBERS = {
+    className: 'number',
+    begin: MATHEMATICA_NUMBER_RE,
+  };
+
+  const SYMBOL_RE = /[a-zA-Z$][a-zA-Z0-9$]*/;
+  const SYMBOLS = {
+    className: 'symbol',
+    begin: SYMBOL_RE,
+    keywords: {
+      className: 'builtin-symbol',
+      $pattern: SYMBOL_RE,
+      'builtin-symbol' : Mathematica.SYSTEM_SYMBOLS.join(" ")
+    },
+  };
+
+  const NAMED_CHARACTER = {
+    className: 'named-character',
+    begin: /\\\[[$a-zA-Z][$a-zA-Z0-9]+]/
+  };
+
+  const OPERATORS = {
+    className: 'operator',
+    begin: /[+\-*/,;.:@~=><&|_`'^?!%]+/
+  };
+
+  const PATTERNS = {
+    className: 'pattern',
+    begin: /([a-zA-Z$][a-zA-Z0-9$]*)?_+([a-zA-Z$][a-zA-Z0-9$]*)?/,
```

Comment:
This is too generic to score (can match almost any identifier in any language)... so I think it also needs to be relevance 0... if you wanted some relevance here you might want a second variant that includes an official list of mathematiac types (how many are there?), ie something like:

```js
booger_booger // 0 relevance
booger_Real // 1 relevance
```

Reply:
No, I agree that this is should also get the relevance of 0. Actually, I don't think there are any outstanding things in short Mathematica snippets that would identify the language. Comments, `(* ... *)`, would be useful but no one really uses them in short snippets especially if you have markdown around it to explain things.

Reply:
None of those 1000 keywords? :-)  But yeah short snippets are worst case scenario for ALL languages... comments definitely count, but yeah not lots of points there for small snippets.  SE auto-hinting for tags is super useful here, though it falls down when a post has more than one tag that has an auto-hinting (which is something SE could fix).

You always get the best results by manually specifying language... and SE can get close to that on MANY posts with auto-hinting to manually specify the language.  Perhaps on the "Mathematica site" (it has it's own site right?) you could even add in an extra bias towards Mathematica... it wouldn't be hard to do.  Of course now you'll get false positives in the other direction but if 99% of the code posted in Mathematica that might be an OK tradeoff.

Reply:
On mathematica.SE our code-blocks always default to "mathematica". 99% of the code is Mathematica anyway and we have this behaviour for a long time. On StackOverflow, there is no support for Mathematica highlighting since every question about it usually gets either closed or moved to our side. Therefore, I'm not really worried about having no rule specific enough to detect Mathematica.

Reply:
Sounds like a win win then. :)